docs/sdkx-cocos2dx/delegates.mdx

```diff
@@ -0,0 +1,320 @@
+---
+sidebar_position: 110
+title: Helpshift Delegates
+description: "The Helpshift SDK provides delegate callbacks to help app developers track a user's activities within the help section."
+---
+
+import { Admonition, Intro } from "@site/src/components/forDocs";
+
+# Helpshift Delegates {#delegates}
+
+<Intro>
+
+The Helpshift SDK provides delegate callbacks to help app developers track a user's activities within the help section.
+
+</Intro>
+
+<Admonition type="info" title="Note">
+
+All the public APIs in the SDK should be called after initializing the SDK via [install API](/sdkx-cocos2dx/getting-started#initializing)
+
+</Admonition>
+
+## Helpshift Events Listener/Delegates implementation {#helpshift-delegates-implementation}
+
+You can set `HelpshiftEventsListener` by calling the `HelpshiftCocos2dx::setHelpshiftEventsListener()` method.
+
+For example:
+
+```cpp
+
+#include "HelpshiftEvent.h"
+
+void handleAuthFailureEvent(HSAuthenticationFailureReason reason) {
+    // your code here
+    // ...
+}
+
+void handleHelpshiftEvent(const char *eventName, cocos2d::ValueMap &eventData) {
+
+    if (strcmp(eventName, SDK_SESSION_STARTED) == 0) {
+        // your code here
+        //...
+    }
+    if (strcmp(eventName, RECEIVED_UNREAD_MESSAGE_COUNT) == 0) {
+        // your code here
+        //...
+    }
+    // ... and so on
+    // list of events given below
+}
+
+// init method of your UI
+bool HelloWorld::init()
+{
+    //////////////////////////////
+    // 1. super init first
+    if ( !Scene::init() )
+    {
+        return false;
+    }
+
+    HelpshiftCocos2dx::setHelpshiftEventsListener(handleHelpshiftEvent,
```

Comment:
there should be code for install call before this so that it clear

Reply:
@sumeet-helpshift We dont allow install call in android from cocos layer.